packages/experimental/kai/src/playwright/app-manager.ts

```diff
@@ -0,0 +1,98 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import type { Context as MochaContext } from 'mocha';
+import type { ConsoleMessage, Page } from 'playwright';
+
+import { sleep, synchronized, Trigger } from '@dxos/async';
+import { setupPage } from '@dxos/test';
+
+// TODO(wittjosiah): Get this from executor.
+const BASE_URL = 'http://localhost:4200';
+
+export class AppManager {
+  page!: Page;
+
+  private _initialized = false;
+  private _invitationCode = new Trigger<string>();
```

Comment:
reaonly

Reply:
can each trigger be fired multiple times? if not, then making it readonly would restrict this to doing a single invitation